File: front-end/src/application/repository/truckRepository.ts
import { Truck } from "../../domain/entity/truck";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from "../../utils/ull-uuid";

export interface TruckRepository {
    save: (truck: Truck) => Promise<Either<DataError, Truck>>;
    
    find: (id: UllUUID) => Promise<Either<DataError, Truck>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Truck[]>>;
    
    update: (id: UllUUID, truck: Truck) => Promise<Either<DataError, Truck>>;
    
    delete: (id: UllUUID) => Promise<Either<DataError, Truck>>;
    
    deleteAll: () => Promise<Either<DataError, Truck[]>>;
}

File: back-end/src/main/java/com/trucks/application/port/TruckRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import com.trucks.domain.entity.Truck;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public interface TruckRepository {

    Truck save(Truck truck);

    Optional<Truck> find(UUID id);
    
    Page<Truck> findAll(PageRequest pageRequest);

    Truck update(UUID id, Truck truck);
    
    Truck delete(UUID id);

    List<Truck> deleteAll();
}
